src/client/services/conversation/conversation.js

```diff
@@ -89,6 +92,46 @@ var ConversationService = SparkBase.extend(
     this.encryptionDisabled = false;
   },
 
+  getExifData: function getExifData(file, buf) {
+    return new Promise(function(resolve) {
+      if (file && file.image && file.mimeType === 'image/jpeg') {
+        new ExifImage({ image : buf }, function (error, exifData) {
+          if (error) {
+            this.logger.info('conversation: error with image rotation ', error.message);
+          }
+          else {
+            if (exifData) {
+              file.image.orientation = exifData.image.Orientation;
+            }
+          }
+          resolve(buf);
+        });
+      }
+      else {
+        resolve(buf);
+      }
+    }.bind(this))
+  },
+
+  fixImageOrientation: function fixImageOrientation(file) {
```

Comment:
For 1 and 2, isn't the file pretty much immediately passed through ShareActivity#add()? It seems like that's the better entry point here. I'm really against the idea of adding a method like this to the root of the conversation code.